features/FEATURE_LWIP/TESTS/mbedmicro-net/udp_echo/main.cpp

```diff
@@ -25,19 +29,40 @@ namespace {
     char rx_buffer[MBED_CFG_UDP_CLIENT_ECHO_BUFFER_SIZE] = {0};
     const char ASCII_MAX = '~' - ' ';
     const int ECHO_LOOPS = 16;
+    char uuid[GREENTEA_UUID_LENGTH] = {0};
 }
 
-void prep_buffer(char *tx_buffer, size_t tx_size) {
-    for (size_t i=0; i<tx_size; ++i) {
+// Creates a buffer that contains the test's UUID in the first part of the contents
+// so the output can be associated with individual test runs. The rest of the
+// buffer is filled with random data so it is unique within the CURRENT test run.
+//
+// Ex. A test with UUID of `33e5002c-9722-4685-817a-709cc69c4701` would have a
+// buffer filled with something like `33e5002c-9722-4685-817a-709cc69c4701 12594387`
+// where `33e5002c-9722-4685-817a-709cc69c4701` is the UUID and `12594387` is the random data
+void prep_buffer(char *uuid, char *tx_buffer, size_t tx_size) {
+    size_t i = 0;
+
+    memcpy(tx_buffer, uuid, strlen(uuid));
+    i += strlen(uuid);
+
+    tx_buffer[i++] = ' ';
+
+    for (; i<tx_size; ++i) {
         tx_buffer[i] = (rand() % 10) + '0';
     }
 }
 
-int main() {
-    GREENTEA_SETUP(60, "udp_echo");
-
+void test_udp_echo() {
     EthernetInterface eth;
-    eth.connect();
+
+    int err = eth.connect();
+    TEST_ASSERT_EQUAL(0, err);
```

Comment:
Why do we have TEST_ASSERT_EQUAL(0, err) both before and after the if statement ?